code/modules/projectiles/guns/projectile.dm

```diff
@@ -148,6 +150,9 @@
 				if(ammo_magazine)
 					user << SPAN_WARNING("[src] already has a magazine loaded.") //already a magazine here
 					return
+				if(!(AM.mag_well & mag_well))
+					user << SPAN_WARNING("[src] can't fit this type of magazine.") //wrong magazine
```

Comment:
Do not add new `<<` to the code. Use `to_chat`.

Reply:
Should be done